packages/indexer/src/indexers/autocrat/autocrat-v1_0-indexer.ts

```diff
@@ -0,0 +1,217 @@
+import { IDL, AutocratV1 } from "@metadaoproject/futarchy-sdk/lib/idl/autocrat_v1";
+import { InstructionIndexer } from "../instruction-indexer";
+import { logger } from "../../logger";
+import { ConditionalVaultIndexer, DaoIndexer, MarketIndexer, ProposalIndexer } from "../program-idl-data-indexer";
+import { rpcReadClient, indexerReadClient } from "../../connection";
+import { usingDb, schema, sql } from "@metadaoproject/indexer-db";
+import { Dao, DaoAggregate } from "@metadaoproject/futarchy-sdk/lib/types";
+import { Err, Ok } from "../../match";
+import { ConditionalVault } from "@metadaoproject/futarchy-ts";
+
+const AUTOCRAT_V1_0 = AUTOCRAT_VERSIONS[AUTOCRAT_VERSIONS.length];
+
+export enum AutocratV1_0DaoIndexerError {
+  GeneralError = "GeneralError",
+}
+
+if (AUTOCRAT_V1_0.label !== "V1") {
+  const error = new Error(`Mistook autocrat ${AUTOCRAT_V1_0.label} for V1`);
+  logger.error(error.message);
+  throw error;
+}
+
+export const AutocratV0_1Indexer: InstructionIndexer<AutocratV1> = {
+  PROGRAM_NAME: "AutocratV1.0",
+  PROGRAM_ID: AUTOCRAT_V1_0.programId.toBase58(),
+  PROGRAM_IDL: IDL,
+  indexInstruction: async (dbTx, txIdx, txRes, ixIdx, ix) => {
+    return Ok;
+  },
+};
+
+// Order for indexing is as follows
+// 1. Need to fetch tokens from daos and
+// insert them into the database
+// 2. After we have the tokens in the database
+// we can insert the dao (which we fetched)
+// 3. After we have the dao, we fetch conditional vaults
+// to generate the p/f tokens (base, quote) and insert them
+// 4. After the tokens are inserted we can then insert the vault
+// 5. And from the vaults insert we can insert token accounts
+// from conditional vaults
+// 6. We then fetch the proposals and insert the markets
+// 7. Once markets are inserted then we can insert the proposal
+
+export const AutocratV1_0DaoIndexer: DaoIndexer = {
+  index: async() => {
+    try {
+      // Fetches all daos from the database
+      const daos: DaoAggregate[] = await indexerReadClient.fetchAllDaos();
+
+      // Check to see if we have database values
+      if(daos == undefined || !daos.length){
+        console.warn(`Database seems to be empty, working off chain exclusively.`);
+      }
+      // Map the database dao keys (TODO: We could argue a simple query is better than leveraging the sdk...)
+      const daoKeys: string[] = daos.flatMap((dao) => dao.daos.flatMap((daoKey) => daoKey.publicKey.toString()))
+
+      // Only working for this version (v1.0)
+      // TODO: Review this so we match on the correct key or it has the f=ing key so we're good
+      const onChainDaos = await rpcReadClient.fetchAllDaos();
+
+      // Check to see if the on chain data returns
+      if(onChainDaos == undefined || !onChainDaos.length){
+        console.error(`No on chain data discovered for program, discontinuing.`);
+        // TODO: Throw but we want to keep cycling
+        return Err({ type: AutocratV1_0DaoIndexerError.GeneralError });
```

Comment:
return error makes sense here instead of throw. BUT it would be nice to have a more specific error type here but not necessary now

Reply:
Indeed